big_tests/tests/mam_helper.erl

```diff
@@ -277,56 +284,74 @@ maybe_with_elem(BWithJID) ->
         name = <<"with">>,
         children = [#xmlcdata{content = BWithJID}]}.
 
+stanza_metadata_request() ->
+    escalus_stanza:iq(<<"get">>,
+                      [#xmlel{name = <<"metadata">>,
+                              attrs = [{<<"xmlns">>, mam_ns_binary_v06()}]}]).
+
 %% An optional 'queryid' attribute allows the client to match results to
 %% a certain query.
 stanza_archive_request(P, QueryId) ->
-    stanza_lookup_messages_iq(P, QueryId,
-                              undefined, undefined,
-                              undefined, undefined, undefined).
+    stanza_lookup_messages_iq(P, QueryId, undefined, undefined,
+                              undefined, undefined, undefined, undefined, undefined).
 
 stanza_date_range_archive_request(P) ->
     stanza_lookup_messages_iq(P, undefined,
                               "2010-06-07T00:00:00Z", "2010-07-07T13:23:54Z",
-                              undefined, undefined, undefined).
+                              undefined, undefined, undefined, undefined, undefined).
 
 stanza_date_range_archive_request_not_empty(P, Start, Stop) ->
     stanza_lookup_messages_iq(P, undefined,
                               Start, Stop,
-                              undefined, undefined, undefined).
+                              undefined, undefined, undefined, undefined, undefined).
 
 stanza_limit_archive_request(P) ->
     stanza_lookup_messages_iq(P, undefined, "2010-08-07T00:00:00Z",
-                              undefined, undefined, #rsm_in{max=10}, undefined).
+                              undefined, undefined, #rsm_in{max=10},
+                              undefined, undefined, undefined).
 
 stanza_page_archive_request(P, QueryId, RSM) ->
-    stanza_lookup_messages_iq(P, QueryId, undefined, undefined, undefined, RSM, undefined).
+    stanza_lookup_messages_iq(P, QueryId, undefined, undefined, undefined,
+                              RSM, undefined, undefined, undefined).
+
+stanza_flip_page_archive_request(P, QueryId, RSM) ->
+    stanza_lookup_messages_iq(P, QueryId, undefined, undefined, undefined,
+                              RSM, undefined, true, undefined).
 
 stanza_filtered_by_jid_request(P, BWithJID) ->
     stanza_lookup_messages_iq(P, undefined, undefined,
-                              undefined, BWithJID, undefined, undefined).
+                              undefined, BWithJID, undefined, undefined, undefined, undefined).
 
 stanza_text_search_archive_request(P, QueryId, TextSearch) ->
     stanza_lookup_messages_iq(P, QueryId,
                               undefined, undefined,
-                              undefined, undefined, TextSearch).
+                              undefined, undefined, TextSearch, undefined, undefined).
```

Comment:
I am wondering if we should use a map here at this point (and no undefined needed in this case).